rego/hdfs.rego

```diff
@@ -12,21 +13,22 @@ allow if {
     action_sufficient_for_operation(acl.action, input.operationName)
 }
 
-# HDFS group mapper (this returns a list of strings)
-groups := {group |
-    raw = groups_for_user[input.username][_]
-    # Keycloak groups have trailing slashes
-    group := trim_prefix(raw, "/")
+# Identity mentions the (long) userName or shortUsername explicitly
+matches_identity(identity) if {
+    identity in {
+        concat("", ["user:", input.callerUgi.userName]),
+        concat("", ["shortUser:", input.callerUgi.shortUserName])
+    }
 }
 
-# Identity mentions the (long) userName explicitly
+# Identity regex matches the (long) userName
 matches_identity(identity) if {
-    identity == concat("", ["user:", input.callerUgi.userName])
+    match_entire(identity, concat("", ["userRegex:", input.callerUgi.userName]))
 }
 
-# Identity mentions the shortUserName explicitly
+# Identity regex matches the shortUsername
 matches_identity(identity) if {
-    identity == concat("", ["shortUser:", input.callerUgi.shortUserName])
+    match_entire(identity, concat("", ["shortUserRegex:", input.callerUgi.shortUserName]))
```

Comment:
There is no test case for this line.

Reply:
Isn't https://github.com/stackabletech/hdfs-utils/blob/00bdb221ec776df6b60292e06f86cecfc173654a/rego/hdfs.rego#L237 and https://github.com/stackabletech/hdfs-utils/blob/7baeecd5454584fe521bc08f9d5818e7674230dc/rego/hdfs_test.rego#L144 a test-case for this?

Reply:
No, because this ACL entry also matches:

https://github.com/stackabletech/hdfs-utils/blob/7baeecd5454584fe521bc08f9d5818e7674230dc/rego/hdfs.rego#L231-L235

And therefore the test passes even if you change the line to:

```rego
match_entire(identity, concat("", ["nonsense:", input.callerUgi.shortUserName]))
```

Reply:
Ah I see, good catch! Please have a look at 7828235